### PR TITLE
Add TestxServer, TestxController and TestxDirector documentation

### PR DESCRIPTION
**Pull request by b1zzu**
### Description

Add documentation for `TestxServer`, `TestxController` and `TestxDirector` to `README.md`

Rendered docs here: https://github.com/b1zzu/graphql-testx/blob/add-api-documentation-to-readme/README.md

##### Checklist

- [x] `npm test` passes
- [x] `npm run build` works
- [x] documentation is changed or added

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,8 +1,11 @@
 # Contributing to GraphQL Testx
 
-First of all, thank you for your interest in contributing to the project :heart_eyes::tada:
+First of all, thank you for your interest in contributing to the project
```
Comment (wtrocki) [inline CONTRIBUTING.md]:
```suggestion
Thank you for your interest in contributing to the GraphQL-TestX project
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
+:heart_eyes::tada:
 
-The following is a set of guidelines for contributing to **GraphQL Testx**. These are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.
+The following is a set of guidelines for contributing to **GraphQL Testx**.
```
Comment (wtrocki) [inline CONTRIBUTING.md]:
```suggestion
Before contributing please read an set of the guidelines for contributing to **GraphQL Testx**:
```

File: README.md [hunk 3]
```diff
@@ -1,19 +1,16 @@
 # graphql-testx
 
-<p align="center">
-  <img width="128" src="https://user-images.githubusercontent.com/7964685/69164244-eeea1080-0aef-11ea-86f0-f34debab1682.png">
-</p>
-
-graphql-testx is a full-featured GraphQL server, based on Graphback and Apollo Server. With the minimum configuration required, you have a server ready for testing GraphQL client applications or libraries.
-Unlike mocking alternatives, graphql-testx offers persistent data between queries and mutation using in-memory SQLite database.
+graphql-testx is a full-featured GraphQL server, based on Graphback and Apollo
```
Comment (wtrocki) [inline README.md]: I would add links to the projects

File: README.md [hunk 4]
```diff
@@ -51,87 +48,328 @@ console.log(`Running on ${await server.httpUrl()}`);
...
+
+#### `httpUrl(): Promise<string>`
+
+Return the http url to the `TestxController` API.
+
+#### `stop(): Promise<void>`
+
+Stop and close the `TestxController` and the `TestxServer`
+
+### `TestxDirector`
```
Comment (wtrocki) [inline README.md:364]: I'm not getting this. Maybe explaining it better.

File: README.md [hunk 5]
```diff
@@ -51,87 +48,328 @@ console.log(`Running on ${await server.httpUrl()}`);
...
+
+Client API for the `TestxController` that implements all `TestxServer` methods.
+
+Is used in combination wit the `TestxController`.
+
+```ts
+import { TestxDirector } from "graphql-testx/dist/src/TestxDirector";
```
Comment (wtrocki) [inline README.md:371]: This should be important. This type of import should not be used

**Review by wtrocki — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (b1zzu) [reply README.md:371]: I know this type of import is a bad practice, but the alternative would be to create a separate package for the TestxDirector, the reason is that the director is used inside browsers, if we would import the director from the index.ts then tools like webpack would also try to import all testx modules, which will import nodejs modules

Comment (b1zzu) [reply README.md:371]: If it's not understandable I can send you an example

File: README.md (hunk 4 above)
Comment (b1zzu) [reply README.md:364]: What do you mean? that's just the name of the class, how could I explain it better?

**Comment by b1zzu** [comment]
I'm going to merge this PR, but I keep the conversations open for future discussions

**Pull request merged**